pkg/pusher/pusher_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pusher_test
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/localstore"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/pusher"
+	"github.com/ethersphere/bee/pkg/pushsync"
+	pushsyncmock "github.com/ethersphere/bee/pkg/pushsync/mock"
+	"github.com/ethersphere/bee/pkg/pushsync/pb"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/topology/mock"
+)
+
+type Store struct {
+	storage.Storer
+	ModeSet   map[string]storage.ModeSet
+	modeSetMu *sync.Mutex
+}
+
+func (s Store) Set(ctx context.Context, mode storage.ModeSet, addrs ...swarm.Address) error {
+	s.modeSetMu.Lock()
+	defer s.modeSetMu.Unlock()
+	for _, addr := range addrs {
+		s.ModeSet[addr.String()] = mode
+	}
+	return nil
+}
+
+func TestSendChunkToPushSync(t *testing.T) {
+	chunk := createChunk(t)
+
+	// create a trigger  and a closestpeer
+	triggerPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
+	closestPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
+
+	pushSyncService := pushsyncmock.New(func(ctx context.Context, address swarm.Address, chunk swarm.Chunk) (*pb.Receipt, error) {
+		receipt := &pb.Receipt{
```

Comment:
can `return &pb....`

Reply:
yeah but doesn't this looks readable and clean? if you  insist i will change